.github/workflows/r_nightly.yml

```diff
@@ -65,54 +68,58 @@ jobs:
+            echo "GOT_FILES=false" >> $GITHUB_ENV
+          fi
       - name: Build Repository
+        if: env.GOT_FILES == 'true'
         shell: Rscript {0}
         run: |
+          repo_root <- "repo"
           art_path <- list.files("binaries",
-              recursive = TRUE,
-              include.dirs = TRUE,
-              pattern = "r-nightly-packages$",
-              full.names = TRUE
+            recursive = TRUE,
+            include.dirs = TRUE,
+            pattern = "r-nightly-packages$",
+            full.names = TRUE
           )
 
-          pkgs <- list.files(art_path, pattern = "r-pkg_*")
-          src_i <- grep("r-pkg_src", pkgs)
-          src_pkg <- pkgs[src_i]
-          pkgs <- pkgs[-src_i]
-          libs <- list.files(art_path, pattern = "r-libarrow*")
+          current_path <- list.files(art_path, full.names = TRUE, recursive = TRUE)
+          files <- sub("r-(pkg|lib)", repo_root, current_path)
+          new_paths <- gsub("__", "/", files)
+          new_paths <- sub(art_path, ".", new_paths)
+          dirs <- dirname(new_paths)
+          dir_result <- sapply(dirs, dir.create, recursive = TRUE)
 
-          new_names <- sub("r-pkg_", "", pkgs, fixed = T)
-          matches <- regmatches(new_names, regexec("(([a-z]+)-[\\.a-zA-Z0-9]+)_(\\d\\.\\d)-(arrow.+)$", new_names))
-
-          dir.create("repo/src/contrib", recursive = TRUE)
-          file.copy(paste0(art_path, "/", src_pkg), paste0("repo/src/contrib/", sub("r-pkg_src-", "", src_pkg)))
-          tools::write_PACKAGES("repo/src/contrib", type = "source", verbose = TRUE)
+          if (!all(dir_result)) {
+            stop("There was an issue while creating the folders!")
+          }
 
-          for (match in matches) {
-              path <- paste0("repo/bin/", match[[3]], "/contrib/", match[[4]])
-              path <- sub("macos", "macosx", path)
-              dir.create(path, recursive = TRUE)
-              file <- list.files(art_path, pattern = paste0(match[[1]], "*"), full.names = TRUE)
+          copy_result <- file.copy(current_path, new_paths)
 
-              file.copy(file, paste0(path, "/", match[[5]]))
-              tools::write_PACKAGES(path, type = paste0(substring(match[[3]], 1, 3), ".binary"), verbose = TRUE)
+          if (!all(copy_result)) {
+            stop("There was an issue while copying the files!")
           }
 
-          lib_names <- sub("r-libarrow-", "", libs)
-          lib_match <- regmatches(lib_names, regexec("([[:alpha:]]+)-(.+)", lib_names))
+          tools::write_PACKAGES(file.path(repo_root, "src/contrib"), type = "source", verbose = TRUE)
 
-          dir.create("repo/libarrow/bin/windows", recursive = TRUE)
-          dir.create("repo/libarrow/bin/centos-7", recursive = TRUE)
-          dir.create("repo/libarrow/bin/ubuntu-18.04", recursive = TRUE)
+          repo_dirs <- list.dirs("repo")
+          pkg_dirs <- repo_dirs[grepl(".+contrib\\/\\d.+", repo_dirs)]

```

Comment:
```suggestion
          pkg_dirs <- grep(".+contrib\\/\\d.+", repo_dirs, value = TRUE)

```

Reply:
❓ ❗  how did I miss that argument until now 🤯